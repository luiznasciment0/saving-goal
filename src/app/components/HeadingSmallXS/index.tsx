import Typography, { ColorProps } from 'app/components/Typography'

interface HeadingSmallXSProps {
  children: React.ReactNode
  color?: ColorProps
}

const HeadingSmallXS = ({
  children,
  color = 'primaryBlack'
}: HeadingSmallXSProps) => (
  <Typography as="h2" family="secondary" size="xlarge" color={color}>
    {children}
  </Typography>
)

export default HeadingSmallXS
