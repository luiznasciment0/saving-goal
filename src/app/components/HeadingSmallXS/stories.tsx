import { Story, Meta } from '@storybook/react/types-6-0'
import HeadingSmallXS from '.'

export default {
  title: 'HeadingSmallXS',
  component: HeadingSmallXS,
  argTypes: {
    children: {
      type: 'string'
    }
  }
} as Meta

export const Default: Story = (args) => (
  <HeadingSmallXS>{args.children}</HeadingSmallXS>
)

Default.args = {
  children: 'Teste'
}
