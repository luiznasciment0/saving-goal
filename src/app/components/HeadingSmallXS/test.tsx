import { render, screen } from '@testing-library/react'

import HeadingSmallXS from '.'

describe('<HeadingSmallXS />', () => {
  it('should render the HeadingSmallXS and its styles', () => {
    const { container } = render(<HeadingSmallXS>Test</HeadingSmallXS>)

    expect(screen.getByRole('heading', { name: /test/i })).toHaveStyle({
      'font-size': '2rem',
      'line-height': '2.4rem',
      'font-weight': '500',
      'font-family': `'Rubik'`,
      color: '#1e2a32'
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
