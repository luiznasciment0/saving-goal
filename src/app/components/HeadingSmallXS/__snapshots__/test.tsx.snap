// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<HeadingSmallXS /> should render the HeadingSmallXS and its styles 1`
] = `
.c0 {
  font-size: 2rem;
  line-height: 2.4rem;
  font-family: 'Rubik';
  color: #1e2a32;
  font-weight: 500;
}

<h2
  class="c0"
  color="primaryBlack"
>
  Teste
</h2>
`
